src/main/java/copado/onpremise/job/OnPremiseDeploymentJob.java

```diff
@@ -76,48 +94,207 @@ public void execute() {
             gitTMP = createTemporalGitPath();
             deployZipFileTMPDir = createTemporalDeployZipPath();
 
-            GitSession git = gitService.cloneRepo(gitTMP);
+            GitSession git = gitService.cloneRepo(gitTMP, gitCredentialService.getCredentialsForMainRepository());
```

Comment:
I would appreciate some tests on this class

Reply:
* Git credential service: This service will be fully implemented by the customer we only provide as product an interface.

The current implementation is just an example, we can add some test in the future but right now I think that make not so much sense, right?